csds-client/client/v3/client.go

```diff
@@ -198,20 +202,21 @@ func (c *ClientV3) doRequest(streamClientStatus csdspb_v3.ClientStatusDiscoveryS
 }
 
 // parseConfigStatus parses each xds config status to string
-func parseConfigStatus(xdsConfig []*csdspb_v3.PerXdsConfig) []string {
+func parseConfigStatus(xdsConfig []*csdspb_v3.ClientConfig_GenericXdsConfig) []string {
 	var configStatus []string
-	for _, perXdsConfig := range xdsConfig {
-		status := perXdsConfig.GetStatus().String()
+	for _, genericXdsConfig := range xdsConfig {
+		status := genericXdsConfig.GetConfigStatus().String()
 		var xds string
-		if perXdsConfig.GetClusterConfig() != nil {
+		switch genericXdsConfig.GetTypeUrl() {
+		case "type.googleapis.com/envoy.config.cluster.v3.Cluster":
 			xds = "CDS"
-		} else if perXdsConfig.GetListenerConfig() != nil {
+		case "type.googleapis.com/envoy.config.listener.v3.Listener":
 			xds = "LDS"
-		} else if perXdsConfig.GetRouteConfig() != nil {
+		case "type.googleapis.com/envoy.config.route.v3.RouteConfiguration":
 			xds = "RDS"
-		} else if perXdsConfig.GetScopedRouteConfig() != nil {
+		case "type.googleapis.com/envoy.config.route.v3.ScopedRouteConfiguration":
 			xds = "SRDS"
-		} else if perXdsConfig.GetEndpointConfig() != nil {
+		case "type.googleapis.com/envoy.config.endpoint.v3.ClusterLoadAssignment":
```

Comment:
Consider erroring out/emitting a warning on non-supported type.

Reply:
Added the handling for unsupported XDS type.